### ISSUE TITLE
DIRT 5 (1038250)

### ISSUE THREAD

**Issue by cow-killer** [ValveSoftware/Proton#4498]
# Compatibility Report
- Name of the game with compatibility issues: DIRT 5
- Steam AppID of the game: 1038250

## System Information
- GPU: GTX 1660
- Driver/LLVM version: 455.45.01
- Kernel version: 5.9.14-zen1-1-zen
- Link to full system information report: [Gist](https://gist.github.com/cow-killer/b4384f94016cae92fd66fcfe2fe328ff)
- Proton version: 5.13-4, Proton Experimental

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

Proton log [here](https://gist.github.com/cow-killer/92efa9b70f86a3014411a7f5934e923a).

## Symptoms 
Nothing appears on screen while the game is running. Tried vanilla Proton 5.13-4, Proton Experimental, and Proton GE 5.21-1. Pretty sure there isn't any sort of anti-cheat installed with this, so I imagine it's a DRM issue?

Sorry if this has already been reported; based on my previous searches I couldn't find any existing issues of this.

**Comment by F1xteR** [comment]
Well on protondb its pretty borked and no currently known workarounds work for the game, guess you have to wait for the devs to react.

**Comment by AsciiWolf** [comment]
DIRT 5 was mentioned in the VKD3D-Proton changelog. Did anyone try the game with latest Proton 5.13-5? I wonder if there is any progress...

**Comment by cow-killer** [comment]
Nope, still borked on 5.13-5 and Proton Experimental: https://github.com/HansKristian-Work/vkd3d-proton/issues/475

**Comment by FanderWasTaken** [comment]
I'm currently on Windows 10 and the game here takes around a minute to boot up. Steam says that the game is "running" but nothing shows up for quite a while... I'm not sure if this helps, just something I noticed.

**Comment by cow-killer** [comment]
The game now runs with the latest commit of [vkd3d-proton](https://github.com/HansKristian-Work/vkd3d-proton), see this issue for details: https://github.com/HansKristian-Work/vkd3d-proton/issues/475

I have only tested on NVIDIA though, would be great to see some AMD users try this out and see if it works for them. However, there's no vibration support. Tested on both Xbox Series X and DualSense controllers. Tagging @gofman for this.

**Comment by ivyl** [comment]
> However, there's no vibration support. Tested on both Xbox Series X and DualSense controllers. Tagging @gofman for this.

That's interesting. The game uses xinput for all the compatible controllers and DirectInput for everything else. As far as I am aware vibration with the controllers you have mentioned should work. I've checked X360 with the wireless dongle, Xbox Series X controller model 1914 over the cable (pairing that thing is pain in the neck) and DualSense also over the cable. The rumble is working for me in all cases in about the same way as it does on Windows.

The easiest way for me to test that was to start the tutorial race and just drive into someone else or a wall.

Can you elaborate a bit more on the controllers you are using (device ids) and how you use them? Can you also provide logs with +xinput,+dinput,+rawinput,+hid,+hid_report,+plugplay?

Side note, DualSense has wonky mappings with 5.13-6 as it doesn't include [this patch](https://github.com/ValveSoftware/wine/commit/cffd2a5de32fc19a9ec7856eb0c7c72fdef3dc23) from experimental yet.

**Comment by cow-killer** [comment]
Hmm, evidently I didn't test well enough the first time through. Tested both controllers again and the vibration is there.

**Comment by guustflater** [comment]
game works following https://boilingsteam.com/dirt-5-now-playable-through-proton/

Problem is that the lightning is not correct at least on latest mesa 21.1.0 and proton experimental and GE

**Comment by GaultGame** [comment]
> game works following https://boilingsteam.com/dirt-5-now-playable-through-proton/
> 
> Problem is that the lightning is not correct at least on latest mesa 21.1.0 and proton experimental and GE

Same problem with EndearvousOS and NVIDIA 470.xxx. Lights are either too bright or too dark. The problem seems to be reduced with Proton Experimental and Proton-GE, but it is still almost unplayable.

**Comment by Macribit** [comment]
rendering problems are fixed in latest proton experimental. But the issue still is, that online doesn't work. Can someone else confirm this? just to check if it's not a problem with my setup.

Also you don't need no workarounds anymore.

**Comment by cow-killer** [comment]
Been able to play online since the game was first playable, not sure why someone would have difficulty connecting to the servers.

**Comment by Macribit** [comment]
weird I checked on the forums and this seems also a problem that windows users have quite often. I'm not sure where to start on how to tackle this issue.

**Comment by Macribit** [comment]
Online challenges work, what doesn't work is playing a match concurrently against others. On the main menu is where it says "Online". I get some sort of network connection error.
Does the Online option in the main menu works also for you @cow-killer ??

**Comment by Macribit** [comment]
I checked with another friend, also online doesn't work for her.

**Comment by pzanoni-intel** [comment]
DIRT 5 is not launching on platforms that have the E-core vs P-core thing (Meteor Lake, Lunar Lake, etc): you click play and it crashes, showing a screen mentioning a dump file that you're supposed to e-mail to people. I was able to verify it works on LNL by setting `WINE_CPU_TOPOLOGY=4:0,1,2,3 %command%`.

I don't know what is the appropriate "generic" way to make it work, but perhaps the tip above would help. I would imagine some other game has the same issue and we just need to add DIRT 5 to the list. 

More details in https://gitlab.freedesktop.org/mesa/mesa/-/work_items/15193
